wrapper/src/main/java/software/amazon/jdbc/plugin/failover/FailoverConnectionPlugin.java

```diff
@@ -364,12 +366,10 @@ private void initSettings() {
         TELEMETRY_FAILOVER_ADDITIONAL_TOP_TRACE.getBoolean(this.properties);
   }
 
-  protected void initFailoverMode() {
+  protected void initFailoverMode(final String initialUrl) {
     if (this.rdsUrlType == null) {
       this.failoverMode = FailoverMode.fromValue(FAILOVER_MODE.getString(this.properties));
-      final HostSpec initialHostSpec = this.hostListProviderService.getInitialConnectionHostSpec();
-      this.rdsUrlType = this.rdsHelper.identifyRdsType(initialHostSpec.getHost());
-
+      this.rdsUrlType = this.rdsHelper.identifyRdsType(ConnectionUrlParser.parseHost(initialUrl));
```

Comment:
Why are we changing it? I think initial host works well. Initial host may have a list of urls. Does ConnectionUrlParser.parseHost() supports it?